tensorflow/compiler/jit/xla_platform_info.h

```diff
@@ -74,7 +74,9 @@ class XlaPlatformInfo {
   // If the op associated with this XlaPlatformInfo is placed on an XLA device
   // then device_allocator_ is the xla::Backend's memory allocator.  If the op
   // is placed on a regular CPU or GPU device then device_allocator_ is null.
-  se::DeviceMemoryAllocator* device_allocator_;
+  // The allocator is of unknown provenance; keep it in a shared pointer to
+  // set an artificial refcount of one.
```

Comment:
I'd expect async compilation to always happen on nodes placed on regular CPU and GPU devices, and thus this always being null if async compilation is enabled.  Have you seen otherwise?

Reply:
no, did not. But there is a possibility for a custom allocator, which is a shared_ptr.
That is what triggered making the allocator a shared_ptr as well.